metadata_as_dictionary.go

```diff
@@ -0,0 +1,201 @@
+package ravendb
+
+// IMetadataDictionary describes metadata for a document
+// Note: in Java there's only one subclass of IMetadataDictionary, so for
+// easy porting we alias its name to the implementation
+type IMetadataDictionary = MetadataAsDictionary
```

Comment:
You can safely change it to just `MetadataAsDictionary` after this is done

Reply:
Agreed, added TODO